From 9bb3ac06ed3d1a3456217d478d556b0abf05b585 Mon Sep 17 00:00:00 2001
From: Keith Packard <keithp@keithp.com>
Date: Sun, 13 Apr 2008 21:37:49 -0700
Subject: [PATCH] Only use FOURCC_XVMC when INTEL_XVMC is defined

The XVMC code uses a magic FOURCC code to signal frame updates, but that
code is only defined when the XVMC code is used.
(cherry picked from commit 18ef4158e5574bfc6621b268821532f13e261d13)

diff --git a/src/i830_video.c b/src/i830_video.c
index 5d6c209..91e0e00 100644
--- a/src/i830_video.c
+++ b/src/i830_video.c
@@ -2119,7 +2119,9 @@ i830_display_video(ScrnInfoPtr pScrn, xf86CrtcPtr crtc,
     switch (id) {
     case FOURCC_YV12:
     case FOURCC_I420:
+#ifdef INTEL_XVMC
     case FOURCC_XVMC:
+#endif
 	OVERLAY_DEBUG("YUV420\n");
 #if 0
 	/* set UV vertical phase to -0.25 */
@@ -2376,7 +2378,9 @@ I830PutImage(ScrnInfoPtr pScrn,
     switch (destId) {
     case FOURCC_YV12:
     case FOURCC_I420:
+#ifdef INTEL_XVMC
     case FOURCC_XVMC:
+#endif
 	if (pPriv->rotation & (RR_Rotate_90 | RR_Rotate_270)) {
 	    dstPitch = ((height / 2) + pitchAlignMask) & ~pitchAlignMask;
 	    size = dstPitch * width * 3;
@@ -2489,8 +2493,10 @@ I830PutImage(ScrnInfoPtr pScrn,
 	I830CopyPackedData(pScrn, pPriv, buf, srcPitch, dstPitch, top, left,
 			   nlines, npixels);
 	break;
+#ifdef INTEL_XVMC
     case FOURCC_XVMC:
 	break;
+#endif
     default:
 	break;
     }
-- 
1.5.5.3

